### PR TITLE
Printing checklist report all at once.

### LINKED ISSUE #808: Refactor connectivity check to separate logic from reporting
Right now we directly output the result in stderr.

We want a function that outputs some kind of report object, while the output will be done separately.

### PR DESCRIPTION
**Pull request by fulmicoton**
It avoids the interleaving of log lines.
The printing format is also improved.

Most notably we now display the error details.

Closes #808

### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

File: quickwit-common/src/checklist.rs [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+fn print_checklist(check_list_results: &[(&'static str, anyhow::Result<()>)]) {
+    eprintln!(
+        "\n\n{}\n{}",
+        "---------------------------------------------------".color(GREEN_COLOR),
+        " Checklist ".color(WHITE_COLOR).on_color(GREEN_COLOR)
+    );
+    let mut errors = Vec::new();
+    for (check_item_name, check_item_result) in check_list_results {
+        // for (check_item_name, check_item_result) in &.items {
```
Comment (fmassot) [inline quickwit-common/src/checklist.rs:58]: remove comment

File: quickwit-common/src/checklist.rs [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        .map(|(name, check_item)| async { (&*name, check_item.await) })
+        .collect();
+    let mut check_named_results: Vec<(&'static str, anyhow::Result<()>)> = Vec::new();
+    while let Some((check_item_name, check_item_result)) = check_named_futures.next().await {
+        check_named_results.push((check_item_name, check_item_result));
+    }
+    check_named_results
+}
+
+/// Run a checklist and print out its success and failures on stdout.
```
Comment (fmassot) [inline quickwit-common/src/checklist.rs:101]: `its successes`

File: quickwit-common/src/checklist.rs [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+///
+/// If an error is encounterred, the proccess will exit with exit code 1.
```
Comment (fmassot) [inline quickwit-common/src/checklist.rs:103]: `encountered`

**Review by fmassot — APPROVED** [review]

**Comment by guilload** [comment]
Can we merge this after #836, I've already had my share of bumpy rebases :D

**Comment by fulmicoton** [comment]
@guilload Yes. Let's rebase after #836

File: quickwit-common/src/checklist.rs (hunk 1 above)
Comment (fulmicoton) [reply quickwit-common/src/checklist.rs:58]:
```suggestion
```

File: quickwit-common/src/checklist.rs (hunk 2 above)
Comment (fulmicoton) [reply quickwit-common/src/checklist.rs:101]:
```suggestion
/// Run a checklist and print out its successes and failures on stdout.
```

File: quickwit-common/src/checklist.rs (hunk 3 above)
Comment (fulmicoton) [reply quickwit-common/src/checklist.rs:103]:
```suggestion
/// If an error is encountered, the proccess will exit with exit code 1.
```

**Comment by fmassot** [comment]
gogo rebase & merge :)

File: quickwit-cli/src/index.rs [hunk 4]
```diff
@@ -568,9 +570,18 @@ pub async fn ingest_docs_cli(args: IngestDocsArgs) -> anyhow::Result<()> {
         source_type,
         params,
     };
-    // Override index source config(s) with ad-hoc source config.
-    index_metadata.sources = vec![source_config];
 
+    run_checklist(vec![
```
Comment (guilload) [inline quickwit-cli/src/index.rs:574]: A few lines above, we've already done:
```rust
let metastore_uri_resolver = MetastoreUriResolver::default();
let metastore = metastore_uri_resolver.resolve(&args.metastore_uri).await?;
let mut index_metadata = metastore.index_metadata(&args.index_id).await?;
```

So, performing the connectivity check here is pointless.

File: quickwit-cli/src/service.rs [hunk 5]
```diff
@@ -131,6 +132,16 @@ async fn run_indexer_cli(args: RunIndexerArgs) -> anyhow::Result<()> {
     let index_metadata = metastore.index_metadata(&args.index_id).await?;
     let storage_uri_resolver = quickwit_storage_uri_resolver();
     let storage = storage_uri_resolver.resolve(&index_metadata.index_uri)?;
+    // TODO: check all sources connectivity.
```
Comment (guilload) [inline quickwit-cli/src/service.rs:135]: Same here.

**Pull request merged**